File: src/main/java/com/duxtinto/intellij/plugin/github/codereviews/ui/toolwindow/ToolWindowContent.java
package com.duxtinto.intellij.plugin.github.codereviews.ui.toolwindow;

import javax.annotation.Nonnull;
import javax.swing.*;

public interface ToolWindowContent {
    interface Model {}
    interface View {
        JComponent getContent();
    }
    interface Presenter {
        void setView(@Nonnull View view);
        void displayContent();
    }
}

File: src/main/kotlin/com/duxtinto/intellij/plugin/github/codereviews/ui/toolwindow/ToolWindowContent.kt
package com.duxtinto.intellij.plugin.github.codereviews.ui.toolwindow

import javax.swing.*

interface ToolWindowContent {
    interface Reviewee {
        interface View {
            val content: JComponent
        }
        interface Presenter {
            fun setView(view: Reviewee.View)
        }
    }

    interface Reviewer {
        interface View {
            val content: JComponent
        }
        interface Presenter {
            fun setView(view: Reviewer.View)
        }
    }

    interface Model

    interface Presenter {
        fun displayToolWindow()
    }
}
